### ISSUE TITLE
Lightweight HTML Triplifier

### ISSUE THREAD

**Issue by jmkeil** [SPARQL-Anything/sparql.anything#484]
The package `io.github.sparql-anything.sparql-anything-html` has a heavy storage footprint (>160MB) due to its dependency on `com.microsoft.playwright.driver-bundle`, which basically five times ships Node.js binaries (Windows, Linux, Linux ARM, Mac and Mac ARM). To my understanding, this is needed to run a headless browser that interprets JS in the triplified HTML.

I guess this is not needed in many use cases.

Therefore, I would like to ask you to consider providing an additional lightweight HTML Triplifier that just triplifies the static HTML document. This would result in significantly smaller binaries of upstream projects and would probably also improve the execution time.

**Comment by enridaga** [comment]
This is very much on point. There is a general problem in building an executable that ends up being terribly large because of all possible dependencies on features that a specific user may not need ... I am not sure how to fix this in the short term. One way could be to declare the dependency as "provided" in the module pom and add it in the launchers' build. I wonder whether there is any good practice we can refer to.

**Comment by justin2004** [comment]
it would be nice to make the build more a la carte but let's not clobber the ability to easily run the headless browser. i have had to use this capability several times.

**Comment by jmkeil** [comment]
I do not deny the existence of use cases for the headless variant. Therefore, my request was about a lightweight Triplifier in addition.

**Comment by enridaga** [comment]
@jmkeil I'd like to elaborate a strategy to cope with this issue and I think your case is perfect for that. Can you please clarify how are you using the package? Are you embedding the maven package in your own build? If this is the case, it should be as easy as marking the dependencies as `provided` in the POM and including them only in the runnable builds. Would that work?

**Comment by jmkeil** [comment]
I do not use it yet, but I consider to use it soon, to enable the import of data in several formats into my pipeline based tool. However, it is my concern that the binaries will become quiet large. I thought about either use `sparql-anything-engine` but exclude the maven packages for a few formats that I do not need or which are to large, or in the first place to only use the  `sparql-anything-*` maven packages, I actually want. Having HTML among the supported formats would be nice, but increasing the binaries size by a magnitude wouldn't be worth it to me.

**Comment by luigi-asprino** [comment]
That was the plan, but something went wrong in the between.
The engine module wasn't supposed to import all the triplifiers (that's why we used reflection to instantiate triplifiers).
With [3233751](https://github.com/SPARQL-Anything/sparql.anything/commit/3233751722bfdea540a8ded251b4009c1caeb758) the engine module imports "model", "metadata", "parser" and "archive" which are needed by the FacadeXOpExecutor.

Now, we could add a "libs" package which facilitates importing all the triplifiers.

**Comment by luigi-asprino** [comment]
[3849845](https://github.com/SPARQL-Anything/sparql.anything/commit/38498452bb9d9325e8b93d419dd3a0c39ba7f57a) adds the libs POM artifact referencing all the triplifiers

**Comment by luigi-asprino** [comment]
Of course, the size of the CLI is unchanged (I don't if this can be reduced), but you can import the engine and the individual packages of the formats you need.

**Comment by enridaga** [comment]
I think the problem here is the size of the -html maven package, which is going to include all dependencies regardless whether they are needed to run the task (with now also any23!).

I think the approach would be to mark all external dependencies with scope "provided" and only add them when building the runnable, executable jars. Those can be added to the -libs module or moved to another module, say -dependencies.

**Comment by enridaga** [comment]
I wonder whether the chances in #512 #513 and #514 are sufficient to close this issue.

**Comment by jmkeil** [comment]
Only half of the issue is addressed. SparqlAnything now can be used without packing all triplifiers but only selected ones. However, the second aspect of the issue is not addressed yet: Adding an additional lightweight HTML triplifier, which only makes use of the plain HTML file and therefor would not require a heavyweight combination of libraries and headless browser binaries, but only a tiny DOM parser library. This would already fit for many HTML related use cases, but only causing a small file size footprint.

**Comment by enridaga** [comment]
Also a discussion was opened on this https://github.com/SPARQL-Anything/sparql.anything/discussions/595

I think it's a good idea to decouple the HTML triplifier to produce a smaller executable, but I have to admit there are more pressing issues at the moment. Will keep this open for better times, though.
